src/Processors/QueryPlan/ReadFromMergeTree.cpp

```diff
@@ -441,6 +441,21 @@ Pipe ReadFromMergeTree::spreadMarkRangesAmongStreamsWithOrder(
     if (info.sum_marks == 0)
         return {};
 
+    bool added_prewhere_output = false;
```

Comment:
This requires comment. I see that this is related to projections so I am not sure if I got what is going on here.